lighthouse-core/index.js

```diff
@@ -48,15 +48,27 @@ async function lighthouse(url, flags = {}, configJSON, connection) {
   flags.logLevel = flags.logLevel || 'error';
   log.setLevel(flags.logLevel);
 
-  // Use ConfigParser to generate a valid config file
-  const config = new Config(configJSON, flags);
+  const config = generateConfig(flags, configJSON);
 
   connection = connection || new ChromeProtocol(flags.port, flags.hostname);
 
   // kick off a lighthouse run
   return Runner.run(connection, {url, config});
 }
 
+/**
+ * Generate a Lighthouse Config.
+ * @param {LH.Flags=} flags Optional settings for the Lighthouse run. If present,
+ *   they will override any settings in the config.
+ * @param {LH.Config.Json=} configJson Configuration for the Lighthouse run. If
+ *   not present, the default config is used.
+ * @return {Config}
+ */
+function generateConfig(flags, configJson) {
```

Comment:
the order here is unexpected to me. i know lighthouse's signature matches this order, but ... can we switch to make it the same as the Config constructor